### PR TITLE
Let Light PeerSharing depend on PeerSharing flag

### PR DESCRIPTION
**Pull request by bolt12**

### CODE REVIEW

**Review by karknu — APPROVED** [review]
LGTM

File: ouroboros-network-framework/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 ### Non-breaking changes
 
+- Let light peer sharing depend on the configured peer sharing flag
```
Comment (coot) [inline ouroboros-network-framework/CHANGELOG.md:9]: :grin:
```suggestion
* Let light peer sharing depend on the configured peer sharing flag
```

File: ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs [hunk 2]
```diff
@@ -1263,6 +1266,7 @@ withConnectionManager ConnectionManagerArguments {
                             -- governor.
                     case outboundGovernorInfoChannel of
                       InResponderMode (Just infoChannel) | notifyOutboundGov
+                                                         , PeerSharingEnabled <- ownPeerSharing
```
Comment (coot) [inline ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs:1269]: I'd move this to `notifyOutboundGov` just above, so all conditions are in one place.

File: ouroboros-network/CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,7 @@
   `Ouroboros.Network.Peersharing`
 * Added `PeerSharingAPI` with all the things necessary to run peer sharing.
 * Fix 'any Cold async demotion' test
+- Let light peer sharing depend on the configured peer sharing flag
```
Comment (coot) [inline ouroboros-network/CHANGELOG.md:13]:
```suggestion
* Let light peer sharing depend on the configured peer sharing flag
```

File: ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs [hunk 4]
```diff
@@ -554,6 +554,8 @@ withConnectionManager
     => ConnectionManagerArguments handlerTrace socket peerAddr handle handleError version versionData m
     -> ConnectionHandler  muxMode handlerTrace socket peerAddr handle handleError (version, versionData) m
     -- ^ Callback which runs in a thread dedicated for a given connection.
+    -> PeerSharing
+    -- ^ Own PeerSharing value
```
Comment (coot) [inline ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs:558]: it's surely own value, the we don't have access to remote peer sharing value when we start connection manager.  I'd use `configuration PeerSharing value` instead, then we more precisely communicate where it's coming from.

**Review by coot — CHANGES_REQUESTED** [review]
Just a few minor comments.

If we are at that could you also change the haddocks in `PeerStateActions` for `peerSharing` to just:

> Peer sharing configuration value.

File: ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs (hunk 4 above)
Comment (bolt12) [reply ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs:558]: I left a TODO.

Comment (bolt12) [reply ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs]: we already get it from somewhere else, at the same level so I thought it didn't make much sense to wrap it inside `ConnectionManagerArguments`

File: ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs [hunk 5]
```diff
@@ -1253,9 +1256,9 @@ withConnectionManager ConnectionManagerArguments {
                     let -- True iff the connection can be used by the outbound
                         -- governor.
                         notifyOutboundGov =
-                          case provenance of
-                            Inbound  -> Duplex == dataFlow
-                            Outbound -> False
+                          case (provenance, ownPeerSharing) of
+                            (Inbound, PeerSharingEnabled) -> Duplex == dataFlow
+                            (_, _)                        -> False
```
Comment (coot) [inline ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs:1261]:
```suggestion
                            _                             -> False
```

File: ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs [hunk 6]
```diff
@@ -587,6 +589,7 @@ withConnectionManager ConnectionManagerArguments {
                       ConnectionHandler {
                           connectionHandler
                         }
+                      ownPeerSharing
```
Comment (coot) [inline ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs:592]:
```suggestion
                      peerSharing
```

**Review by coot — APPROVED** [review]
LGTM

**Comment by coot** [comment]
Something is wrong with `ghc-9.8` build on Windows. I re-run it yesterday, but it failed in the same way.

**Pull request merged**